### DISCUSSION TITLE
Quiz GPT 勘誤：Question 2615

### DISCUSSION THREAD

**Discussion by FakeWood** [1101-GS4719-C/Python-Community-GS4719-C#1674] [General]
在集合set的知識點那邊，有一題
[Question 2615](https://ncuedu.tw/learning/chatgpt/cid/1/qid/2615)
這題問的是下列何者正確，
但是C選項跟D選項應該都是對的，
因為Python的集合(set)是一種無序的、元素不重複的容器類型，各元素間沒有順序之分，而且集合不能使用切片操作，因為集合並沒有索引。
再麻煩老師助教更正一下！

![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/105701699/8e3bed7e-6110-4a14-9aea-882e745355cf)

如果選(D)
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/105701699/4db77721-e7cc-489d-9246-ec31a5ef1037)

就錯了！
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/105701699/5b6e65ef-6123-4363-8718-3d70164def9c)

如果選(C)
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/105701699/dd986dda-e54a-4a1d-aead-a2f43d178ea4)

就對了！
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/105701699/a502cae5-713c-474c-b49f-61e01d4fe3fe)

**Comment by 1chooo** [comment]
我在想切片操作的意思是不是像是 `list` 的 `test_list[1, 3]` 的操作，如果是的話，那 `set` 確實沒有切片的操作，如以下的範例程式碼：

```py
test_set = {1, 2, 3, 4, 5}

print(test_set)
print(test_set[1, 3])
```

可以看出第二行能印出來，但是在第三行執行會出錯。

```shell
{1, 2, 3, 4, 5}
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<my_apth>/test.py](https://file+.vscode-resource.vscode-cdn.net/<my_apth>/test.py) in line 3
      [12](file:///<my_apth>/test.py?line=11) test_set = {1, 2, 3, 4, 5}
      [13](file:///<my_apth>/test.py?line=12) print(test_set)
----> [14](file:///<my_apth>/test.py?line=13) print(test_set[1, 3])

TypeError: 'set' object is not subscriptable
```

### 後續
不過我有去查 stackoverflow 其實 `set` 也可以用 slice 的操作。
- 原文連結：[What is a pythonic way of slicing a set?](https://stackoverflow.com/questions/40736681/what-is-a-pythonic-way-of-slicing-a-set)

原文中的解法參考：
```py
import itertools

some_data = [1, 2, 4, 1, 6, 23, 3, 56, 6, 2, 3, 5, 6, 32, 2, 12, 5, 3, 2]
big_set = set(some_data)
small_set = set(itertools.islice(big_set, 5))

print(small_set)    # {32, 1, 2, 3, 4}
```

所以其實 python 中的 `set` 應該還是可以有 slice 操作，不過可能需要搭配其他套件使用就是了！

（大家如果有想法都非常歡迎進一步討論！）
